### PR TITLE
Mostly add syntax highlighting ability

### PR DESCRIPTION
**Pull request by ComFreek**
The syntax highlighter works with Pygments. By chance, I already have an MMT Pygments plugin: https://github.com/ComFreek/mmt-pygments-lexer

So in the future we could post MMT code into the documentation *and* get it syntax-highlighted. As of now, it already pays off for the various Scala snippets in there.

You can see it live as of now here: https://comfreek.github.io/uniformal.github.io/doc/tutorials/applications/getting-started.html

image

### CODE REVIEW

**Comment by ComFreek** [comment]
Getting an e-mail notification from GitHub saying 'pygments' is actually not supported and silently replaced by 'rouge'. Damn, so I guess, at least we've got Scala highlighting now, but MMT highlighting will probably be impossible. (Unless deployment is done via Travis, which doesn't restrict Pygments!)

File: _data/menu.yml [hunk 1]
```diff
@@ -56,8 +56,8 @@
   items:
     - title: 'Introductory Articles'
       url: '/doc/philosophy/intros.html'
-    - title: 'The OMDoc Format'
-      url: '/doc/philosophy/omdoc.html' 
+    - title: 'The OM/doc Format'
```
Comment (tkw1536) [inline _data/menu.yml]: Not sure I like the OM/doc spelling. The name is OMDOC (without any /s)

**Review by tkw1536 — CHANGES_REQUESTED** [review]
Apart from the one spelling problem LGTM

**Comment by tkw1536** [comment]
Pity about pigments not being supported. We could always consider using Travis and use github-pages as deploy only.

File: _data/menu.yml (hunk 1 above)
Comment (kohlhase) [reply _data/menu.yml]: Actually, the name is `OMDoc`.

Comment (ComFreek) [reply _data/menu.yml]: Ah, that was an unintentional mistake of mass regex replacement of `/doc` URLs (relativizing them). Reverted in latest commit.

**Comment by ComFreek** [comment]
> Pity about pigments not being supported. We could always consider using Travis and use github-pages as deploy only.

I now added a working Travis setup in my latest commit to this PR: https://github.com/ComFreek/uniformal.github.io/blob/master/.travis.yml

It works on my fork https://comfreek.github.io/uniformal.github.io/doc/ (modulo some path problems with the CSS and JS files - but since this UniFormal repo is hosted "more absolutely" on uniformal.github.io -- and not uniformal.github.io/blah-my-repo-name -- this should *work* here!)

After merging Travis needs to be given a GitHub API key as an env variable under the name `GITHUB_PAGES_DEPLOYMENT_TOKEN`.

**Edit:** I think paths are more broken than expected. Let me investigate before merging.

**Comment by ComFreek** [comment]
@tkw1536 Finally did it! See https://comfreek.github.io/uniformal.github.io/. Should all be working now!

Remember to set `GITHUB_PAGES_DEPLOYMENT_TOKEN` in Travis and set the pages branch in the repo's settings to `gh-pages` (after the first Travis build succeeded).

**Pull request merged**